lib/presentation/weather/weather_page.dart

```diff
@@ -95,7 +39,32 @@ class _BodyState extends State<_Body> {
   }
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final state = ref.watch(weatherPageStateProvider);
+    final weather = state.when(
+      initial: () => null,
+      success: (weather) => weather,
+      error: (_) => null,
+    );
+
+    ref.listen<WeatherUiState>(
```

Comment:
![memo-badge](https://img.shields.io/badge/review-memo-lightgrey)

> The main difference between them is that, rather than rebuilding the widget/provider if the listened to provider changes, using ref.listen will instead call a custom function.
That can be useful for performing actions when a certain change happens, such as showing a snackbar when an error happens.

https://riverpod.dev/docs/concepts/reading#using-reflisten-to-react-to-a-provider-change